packages/records/lib/models/records.ts

```diff
@@ -355,49 +338,61 @@ export async function markNonCurrentGenerationRecordsAsDeleted(connectionId: num
 }
 
 /**
- * Compute Added Keys
- * @desc for any incoming payload use the provided unique to check against the rows
- * in the database and return the keys that are not in the database
- *
+ * getUpdatedKeys
+ * @desc returns a list of the keys that exist in the records tables but have a different data_hash
  */
-async function getAddedKeys(response: FormattedRecord[], connectionId: number, model: string): Promise<string[]> {
-    const keys: string[] = response.map((record: FormattedRecord) => getUniqueId(record));
+async function getUpdatedKeys(records: FormattedRecord[], connectionId: number, model: string): Promise<string[]> {
```

Comment:
consolidating the logic in a single function since there were called together anyway. It allows not having to make a extra query to figure out which keys are going to be deleted. It is the same `knownKeys` query previously used in `getAddedKeys`